File: Habitica/src/main/java/com/habitrpg/android/habitica/utils/FAQArticleListDeserilializer.java
package com.habitrpg.android.habitica.utils;

import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.habitrpg.android.habitica.models.FAQArticle;

import java.lang.reflect.Type;
import java.util.List;

import io.realm.RealmList;

/**
 * Created by viirus on 22/01/16.
 */
public class FAQArticleListDeserilializer implements JsonDeserializer<List<FAQArticle>> {


    @Override
    public List<FAQArticle> deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {
        RealmList<FAQArticle> vals = new RealmList<>();
        int position = 0;
        for (JsonElement e : json.getAsJsonObject().get("questions").getAsJsonArray()) {
            JsonObject obj = e.getAsJsonObject();
            FAQArticle article = new FAQArticle();
            article.setPosition(position);
            article.setQuestion(obj.get("question").getAsString());
            if (obj.has("android")) {
                article.setAnswer(obj.get("android").getAsString());
            } else {
                article.setAnswer(obj.get("web").getAsString());
            }
            vals.add(article);
            position++;
        }

        return vals;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/utils/FAQArticleListDeserilializer.kt
package com.habitrpg.android.habitica.utils

import com.google.gson.JsonDeserializationContext
import com.google.gson.JsonDeserializer
import com.google.gson.JsonElement
import com.google.gson.JsonParseException
import com.habitrpg.android.habitica.models.FAQArticle
import io.realm.RealmList
import java.lang.reflect.Type

/**
 * Created by viirus on 22/01/16.
 */
class FAQArticleListDeserilializer : JsonDeserializer<List<FAQArticle>> {
    @Throws(JsonParseException::class)
    override fun deserialize(
        json: JsonElement,
        typeOfT: Type,
        context: JsonDeserializationContext,
    ): List<FAQArticle> {
        val articles = RealmList<FAQArticle>()
        for ((position, e) in json.asJsonObject.get("questions").asJsonArray.withIndex()) {
            val obj = e.asJsonObject
            val article = FAQArticle()
            article.position = position
            article.question = obj.get("question").asString
            if (obj.has("android")) {
                article.answer = obj.get("android").asString
            } else {
                article.answer = obj.get("web").asString
            }
            articles.add(article)
        }

        return articles
    }
}
